tests/Unit/Support/Traits/AssertCssTest.php

```diff
@@ -0,0 +1,289 @@
+<?php
+
+namespace Pelago\Tests\Unit\Support\Traits;
+
+use Pelago\Tests\Unit\Support\Traits\Fixtures\TestingAssertCss;
+
+/**
+ * Test case.
+ *
+ * @author Jake Hotson <jake.github@qzdesign.co.uk>
+ */
+class AssertCssTest extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @test
+     */
+    public function getCssNeedleRegExpEscapesAllSpecialCharacters()
+    {
+        $needle = '.\\+*?[^]$(){}=!<>|:-/';
+
+        $result = TestingAssertCss::getCssNeedleRegExp($needle);
+
+        static::assertSame(
+            '/' . preg_quote($needle, '/') . '/',
+            str_replace('\\s*+', '', $result)
```

Comment:
Why the str_replace?

Reply:
Because `$result` also contains inserted regex for optional whitespace, so would not match exactly.  I can introduce an intermediate variable `$resultWithOptionalWhitespaceRemoved` to make this (and similar other cases) clearer...